### DISCUSSION TITLE
GitHub Copilot not getting context from my VS2022 project files

### DISCUSSION THREAD

**Discussion by RayWotton** [community/community#78470] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I am having problems with GitHub Copilot not getting context from my VS2022 project files. My GitHub Copilot (I have a Business connection for GitHub Copilot).
Below are the issues that I have:
***
In GitHub Copilot Chat:
(me)
Please code review the selected text
(there was a References icon that showed the cpp filepath)

GitHub Copilot
I'm sorry, but I can't see the code you've selected. As an AI programming assistant, I don't have direct access to your code editor or the ability to see your selected text. I rely on the code snippets or descriptions you provide me in the chat to understand the specific context and provide relevant assistance.
Please paste the code you want me to review into the chat, and I'll be happy to provide feedback.
***
Using the in Editor Alt+/ I get this error:
me: please code review the selected text
GitHub Copilot: I'm sorry, but I cannot directly review the selected text as I don't have access to it.  However, I can provide you with some general tips... (and so on)
***
Installed:
Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.8.3
VisualStudio.17.Release/17.8.3+34330.188
Microsoft .NET Framework Version 4.8.09032
ASP.NET and Web Tools   17.8.358.6298
ASP.NET and Web Tools
Azure App Service Tools v3.0.0   17.8.358.6298

C# Tools   4.8.0-7.23572.1+7b75981cf3bd520b86ec4ed00ec156c8bc48e4eb
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

GitHub Copilot   1.139.0.0 (v1.139.0.0@4eca0246f)
GitHub Copilot Agent   1.139.0

Microsoft JVM Debugger   1.0
NuGet Package Manager   6.8.0
Test Adapter for Boost.Test   1.0
Test Adapter for Google Test   1.0
TypeScript Tools   17.0.20920.2001

Visual Basic Tools   4.8.0-7.23572.1+7b75981cf3bd520b86ec4ed00ec156c8bc48e4eb
Visual F# Tools   17.8.0-beta.23475.2+10f956e631a1efc0f7f5e49c626c494cd32b1f50
Visual Studio IntelliCode   2.2 AI-assisted development for Visual Studio.
***
I also have JetBrains ReSharper installed:
JetBrains ReSharper 2023.3
Build 233.0.20231203.134406 built on 2023-12-03
ReSharper
2023.3.20231204.3853 (up to date)
ReSharper C++
2023.3.20231204.4527 (up to date)
***
I have tried to uninstall and reinstall several times.  Same exact issues...  Thank you for looking into this!  Have a Great Day!  BTW, If I need to put this somewhere else, please let me know where...  Thank You Again!

**Comment by rheapatel23** [comment]
Hi @RayWotton, thanks for bringing this to our attention our team will look into this. 

Would you be able to share your logs for this:

- The ComponentModelCache folder from %localappdata%\microsoft\visualstudio\17.0(your instance)\ComponentModelCache, which contains files such as Microsoft.VisualStudio.Default.err
- The ExtensionMetadata.mpack file from %localappdata%\microsoft\visualstudio\17.0(your instance)\Extensions\ExtensionMetdata.mpack
- run the tool at https://aka.ms/vscollect and collect the logs placed at %TEMP%\vslogs.zip and zip the entire contents of %temp%\vsgithubcopilotlogs


Thanks, let me know if you have other questions. 

Rhea
Product Manager GitHub Copilot for Visual Studio

**Reply by JamesAllerlei** [reply]
Im having the same issue too. My copilot can’t see the context or read a full script

**Reply by ayim** [reply, marked as answer]
@JamesAllerlei can you confirm that you're still experiencing issues in Visual Studio 2022 and with Copilot Chat?

If so, please use "Report a Problem" in Visual Studio to report a bug
![image](https://github.com/community/community/assets/10951869/3a4799df-f69a-4603-8b94-f0cf7c16f4c9)

**Comment by RayWotton** [comment]
Thank You, @rheapatel23 for looking at this...  I Attached the files (I zipped them as the directories they were in) you asked for...  I hope they arrive well.  Let me know if there is something else you need...
Have a Great Day and Thanks Again!
[ExtensionMetadata.zip](https://github.com/community/community/files/13639539/ExtensionMetadata.zip)
[VSGitHubCopilotLogs.zip](https://github.com/community/community/files/13639540/VSGitHubCopilotLogs.zip)
[vslogs.zip](https://github.com/community/community/files/13639541/vslogs.zip)
[ComponentModelCache.zip](https://github.com/community/community/files/13639542/ComponentModelCache.zip)

**Reply by rkpatel33** [reply]
For color, I have the same problem. In the logs I used to see a line item indicating the extension was building the local document index from which it would pull text for the RAG system, but I do not see that anymore.

**Comment by patricia123rash** [comment]
any answer to solve this I've got the exact same issue

**Reply by emok032** [reply]
I've been getting this for the past few weeks now. Hoping the updates would have fixed it.

**Reply by ayim** [reply]
@patricia123rash @emok032 
Are you using the "#" command to specify the file you want to use as a part of your question?
This is the best way to make sure we have your file context.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
GitHub: ayim | aayim@microsoft.com

**Comment by BobDickinson** [comment]
I've been using CoPilot with VS Code every day for a month on a paid plan, and all of the sudden today in the middle of an editing session it got really, really stupid (giving the same "As an AI programming assistant, I don't have direct access to your code editor or the ability to see your selected text." message).  It's giving very generic answers (and in JavaScript, for a TypeScript project).  Completely useless.  Have tried restarting.

**Reply by ayim** [reply]
Are you using the "#" command to specify the file you want to use as a part of your question?
This is the best way to make sure we have your file context.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
GitHub: ayim | aayim@microsoft.com

**Comment by Randil-Hasanga** [comment]
I've got the same issue, In my case problem is the custom proxy.

**Reply by ayim** [reply]
The custom proxy really shouldn't be affecting the code and context that you're packing as a part of your file.
Are you using the "#" command to specify the file you want to use as a part of your question?
This is the best way to make sure we have your file context.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
GitHub: ayim | aayim@microsoft.com

**Comment by gitthub89** [comment]
Same here, i am even using the #file variable to give it access to the exact files it says it couldnt read the previous prompt and it says it cant read them as an AI model... 👎🏻 
my 2nd prompt after it complained it couldnt read the files:

@workspace #file:MessageComponent.tsx #file:Textbox.tsx #file:ChatBubbles.tsx

response:
"... As for the [MessageComponent.tsx] and [Textbox.tsx] files, without seeing the code in these files, I can't provide specific advice on what changes might be needed"
:(

**Reply by ayim** [reply]
@gitthub89 sounds like you're in VSCode, @Akash1134 will take care of ya.

**Comment by ghost** [comment]
Same here, Github Copilot has been getting worse and worse the last few months. I've been using it since the end of last year, and at that time it was  producing incredibly good suggestions that were context-aware and all. But in the last month nearly all suggestions (be it inline suggestions or through the Copilot Chat) have become useless.

BTW, I'm using it in VSCode.
What I've found out: I am creating a React app with NextJS at the moment, and all the pages are named page.tsx in their corresponding subfolder (that's a NextJS thing). When I try to get the context with Github Copilot using the #file command, it seems to always take only the currently open page.tsx file and not the one I explicitely select. Maybe that could be a problem?

**Reply by ayim** [reply]
I would open a separate issue for VSCode if I were you

**Comment by JamesAllerlei** [comment]
Hi Aaron, yes, I'm still experiencing these  issues with VS Code and
copilot chat but I've been talking to support on email and they have
escalated it to the developers. Is it still worthwhile reporting the bug?

Den mån 25 mars 2024 kl 17:10 skrev Aaron Yim ***@***.***>:

**Reply by ayim** [reply]
"Report a Problem" is more for Visual Studio. If you're already in contact with support about the VSCode issue, that's the right place to be.

**Comment by JamesAllerlei** [comment]
Ok thanks. Sent from my phone. On 26 Mar 2024, at 01:38, Aaron Yim ***@***.***> wrote:﻿
"Report a Problem" is more for Visual Studio. If you're already in contact with support about the VSCode issue, that's the right place to be.

**Comment by JamesAllerlei** [comment]
Ok, thanks Aaron. I will.

Den tis 26 mars 2024 kl 02:32 skrev Aaron Yim ***@***.***>:

**Comment by LogicLemur** [comment]
I have this issue also

**Reply by ayim** [reply]
Hi @LogicLemur.

I'm going to try my best to understand your issue:

To start, have you tried using the "#" command to specify the file you want to use as a part of your question?
This is the best way to make sure we have your file context.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
GitHub: ayim | [aayim@microsoft.com](mailto:aayim@microsoft.com)
